src/route-sync/config/config_test.go

```diff
@@ -1,79 +1,155 @@
 package config_test
 
 import (
+	"io/ioutil"
 	"os"
 	. "route-sync/config"
 
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
+
+	"github.com/onsi/gomega/gbytes"
+
+	cfConfig "code.cloudfoundry.org/route-registrar/config"
+	uaaconfig "code.cloudfoundry.org/uaa-go-client/config"
 )
 
 var _ = Describe("Config", func() {
-	Context("Configuration is set", func() {
+	Context("Reading schema from a file that doesn't exist", func() {
+		It("does not return a config schema", func() {
+			cs, err := NewConfigSchemaFromFile("why-do-you-have-this-file")
+			Expect(err).To(HaveOccurred())
+			Expect(cs).To(BeNil())
+		})
+	})
+	Context("Reading schema from a file", func() {
+		var (
+			configFile *os.File
+		)
 
 		BeforeEach(func() {
-			os.Setenv("ROUTESYNC_NATS_SERVERS", "[{\"Host\": \"10.0.1.8:4222\",\"User\": \"nats\", \"Password\": \"natspass\"}]")
-			os.Setenv("ROUTESYNC_CLOUD_FOUNDRY_APP_DOMAIN_NAME", "apps.cf.example.org")
-			os.Setenv("ROUTESYNC_UAA_API_URL", "https://uaa.cf.example.org")
-			os.Setenv("ROUTESYNC_ROUTING_API_URL", "https://api.cf.example.org")
-			os.Setenv("ROUTESYNC_ROUTING_API_USERNAME", "routeUser")
-			os.Setenv("ROUTESYNC_ROUTING_API_CLIENT_SECRET", "aabbcc")
-			os.Setenv("ROUTESYNC_SKIP_TLS_VERIFICATION", "true")
-			os.Setenv("ROUTESYNC_KUBE_CONFIG_PATH", "~/.config/kube")
+			var err error
+			configFile, err = ioutil.TempFile("", "routesync")
+			if err != nil {
+				panic(err)
+			}
 		})
 
 		AfterEach(func() {
-			os.Unsetenv("ROUTESYNC_NATS_SERVERS")
-			os.Unsetenv("ROUTESYNC_CLOUD_FOUNDRY_APP_DOMAIN_NAME")
-			os.Unsetenv("ROUTESYNC_UAA_API_URL")
-			os.Unsetenv("ROUTESYNC_ROUTING_API_URL")
-			os.Unsetenv("ROUTESYNC_ROUTING_API_USERNAME")
-			os.Unsetenv("ROUTESYNC_ROUTING_API_CLIENT_SECRET")
-			os.Unsetenv("ROUTESYNC_SKIP_TLS_VERIFICATION")
-			os.Unsetenv("ROUTESYNC_KUBE_CONFIG_PATH")
+			configFile.Close()
+			os.Remove(configFile.Name())
 		})
 
-		It("returns a valid config frm the enviornment", func() {
-			c, err := NewConfig()
-
-			Expect(err).NotTo(HaveOccurred())
-			Expect(c.NatsServers).To(HaveLen(1))
-
-			natsServer := c.NatsServers[0]
-			Expect(natsServer.Host).To(Equal("10.0.1.8:4222"))
-			Expect(natsServer.User).To(Equal("nats"))
-			Expect(natsServer.Password).To(Equal("natspass"))
+		Context("with an empty file", func() {
```

Comment:
I don't think we need two test cases here.
If file is there and proper formatted, we should expect `ConfigSchema` to be returns.